spec/requests/rotation_generators_controller_spec.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+RSpec.describe 'RotationGeneratorsController' do
+  shared_context 'when logged in as a roster admin' do
+    let(:admin) { create(:user).tap { |user| create :membership, roster:, user:, admin: true } }
+
+    before { set_user admin }
+  end
+
+  describe 'GET assignments/rotation_generator' do
+    subject(:call) { get "/rosters/#{roster.id}/assignments/rotation_generator" }
+
+    let(:roster) { create :roster }
+    let(:user1) { create(:user).tap { |user| create :membership, roster:, user: } }
+
+    include_context 'when logged in as a roster admin'
+
+    it 'responds successfully' do
+      call
+      expect(response).to be_successful
+    end
+  end
+
+  describe 'POST assignments/generate_rotation' do
+    subject(:submit) { post "/rosters/#{roster.id}/assignments/generate_rotation", params: }
+
+    let(:roster) { create :roster }
+    let(:user1) { create(:user).tap { |user| create :membership, roster:, user: } }
+    let(:start_date) { Date.current.beginning_of_week(:sunday) }
+
+    include_context 'when logged in as a roster admin'
+
+    context 'with valid params' do
+      let(:params) do
+        { assignment_rotation_generator: { start_date: start_date,
+                                           end_date: 1.week.from_now.end_of_week(:sunday),
+                                           starting_user_id: user1.id,
+                                           user_ids: [user1.id, admin.id] } }
+      end
+
+      it 'creates new assignments' do
+        expect { submit }.to change(Assignment, :count).by(2)
+      end
+
```

Comment:
```suggestion

it 'creates new assignments with the correct attributes' do
  submit
  expect(Assignment.last(3).collect(&:attributes)).to contain_exactly(
    a_hash_including(...),
    a_hash_including(...),
    a_hash_including(...)
  )
end

```

So these specs are a good start, however we do want to cover as many edge cases as possible and also cover our implementation of the actual rotation generation process.

I think we should be generating at least 3 assignments against 2 users, and the starting user shouldn't be the first entry in `user_ids` (to test the `.rotate` part), and the end date should end earlier than sunday (so that we can test that the last week will get cut off accordingly).

Check in with me if you need a more in depth explanation on how to set up these cases